server/routes/api/chart.py

```diff
@@ -378,3 +378,164 @@ def geojson(dcid):
+    if os.environ.get('FLASK_ENV') == 'development':
+        with bp.open_resource('../../chart_config.json') as f:
+            raw_config = json.load(f)
+    all_sv = set()
+    sv_denom_mapping = {}
+    for config in chart_config:
+        if 'hasChoropleth' in config and config['hasChoropleth']:
+            # we should only be making choropleths for configs with a single stat var
+            sv = config['statsVars'][0]
+            all_sv.add(sv)
+            denom = ''
+            if 'denominator' in config:
+                denom = config['denominator'][0]
+                all_sv.add(denom)
+            elif 'perCapita' in config and config['perCapita']:
+                # The denominator used for perCapita is total population
+                all_sv.add('Count_Person')
+                denom = 'Count_Person'
+            if sv not in sv_denom_mapping:
+                sv_denom_mapping[sv] = set()
+            sv_denom_mapping[sv].add(denom)
+    return all_sv, sv_denom_mapping
+
+
+def get_data_for_statvar(sv, geos, all_sv_data):
+    """ Gets all the data for a specific stat var with the source series data flattened out
+    
+    Args:
+        sv: the stat var we are returning data for
+        geos: list of place dcids we are returning data for
+        all_sv_data: value of the key placeData in the object received from getStatsAll
+
+    Returns:
+        dictionary of dictionary of date: data, keyed by place dcid
+    """
+    result = {}
+    for geo in geos:
+        result[geo] = dict()
+        if geo in all_sv_data and 'statVarData' in all_sv_data[geo]:
+            sv_data = all_sv_data[geo]['statVarData']
+            if sv not in sv_data:
+                continue
+            sv_data = sv_data[sv]
+            if 'sourceSeries' not in sv_data:
+                continue
+            for source in sv_data['sourceSeries']:
+                result[geo].update(source['val'])
+    return result
+
+
+# TODO: Update method of getting dates to ensure stat var date and denominator date matches
+def get_latest_common_date_for_sv(sv_data):
+    """Finds the latest common date
+    
+    Args:
+        sv_data: dictionary of dictionary of date: data, keyed by place dcid
+
+    Returns:
+        date as a string 
+    """
+    common_dates = None
+    all_dates = set()
+    if not sv_data:
+        return None
+    for place, data in sv_data.items():
+        if common_dates:
+            common_dates = common_dates.intersection(data.keys())
+        else:
+            common_dates = set(data.keys())
+        all_dates.update(data.keys())
+    sv_date = None
+    if common_dates:
+        sorted_dates = sorted(common_dates)
+        if len(sorted_dates) > 0:
+            sv_date = sorted_dates[-1]
+    if not sv_date and all_dates:
+        sorted_dates = sorted(all_dates)
+        if len(sorted_dates) > 0:
+            sv_date = sorted_dates[-1]
+    return sv_date
+
+
+@bp.route('/choroplethdata/<path:dcid>')
```

Comment:
oh, should these functions in choropleth.py and the actual path for this endpoint is /choropleth/data/<path:dcid>?

Reply:
I put it in chart.py because the endpoint is getting choropleth data for choropleth charts (based off of chart config), and it seems like choropleth.py contains endpoints for the choropleth tool. Not sure if we want to keep those two things separated

Reply:
Ah, that's right, i was thinking the choropleth tool. Then this is right.